app/main.py

```
@@ -3071,6 +3324,299 @@ async def batch_optimize_images(
+            target_currency=request.target_currency,
+            source=request.source
+        )
+        if not result:
+            raise HTTPException(status_code=400, detail="Failed to create exchange rate")
+        return schemas.ExchangeRateResponse(**result)
+    except Exception as e:
+        logger.error(f"Error creating exchange rate: {str(e)}")
+        raise HTTPException(status_code=400, detail=str(e))
+    
+@app.post("/exchange-rates/organization/list", response_model=ExchangeRateListResponse, tags=["exchange-rates"])
+async def get_organization_exchange_rates(request: CostFilterRequest):
```

Ruff (0.14.14)

[error] Redefinition of unused `request` from line 7: `request` redefined here

(F811)